### PR TITLE
create-byoc-cluster-azure: Microsoft.Compute/EncryptionAtHost is a feature, not a resource provider

### PR DESCRIPTION
**Pull request by ligfx**
**Old:**

> - Resources: The Azure subscription must be registered for the following resource providers. For more information, see the [Microsoft documentation](https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/resource-providers-and-types).
>   - Microsoft.Compute
>   - Microsoft.ManagedIdentity
>   - Microsoft.Storage
>   - Microsoft.KeyVault
>   - Microsoft.Network
>   - Microsoft.Compute.EncryptionAtHost

**New:**

> - Resources: The Azure subscription must be registered for the following resource providers. For more information, see the [Microsoft documentation](https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/resource-providers-and-types).
>   - Microsoft.Compute
>   - Microsoft.ManagedIdentity
>   - Microsoft.Storage
>   - Microsoft.KeyVault
>   - Microsoft.Network
> - Features: The Azure subscription must be registered for the following features.
>   - Microsoft.Compute/EncryptionAtHost. For more information, see the [Microsoft documentation](https://learn.microsoft.com/en-us/azure/virtual-machines/linux/disks-enable-host-based-encryption-cli#prerequisites).

## Preview

[Create a BYOC on Azure docs](https://deploy-preview-16--rp-cloud.netlify.app/redpanda-cloud/get-started/cluster-types/byoc/create-byoc-cluster-azure/)

### CODE REVIEW

File: modules/get-started/pages/cluster-types/byoc/create-byoc-cluster-azure.adoc [hunk 1]
```diff
@@ -16,7 +16,10 @@
 *** Microsoft.Storage
 *** Microsoft.KeyVault
 *** Microsoft.Network
-*** Microsoft.Compute.EncryptionAtHost
++
+** Features: The Azure subscription must be registered for the following features.
```
Comment (asimms41) [inline modules/get-started/pages/cluster-types/byoc/create-byoc-cluster-azure.adoc:20]:
```suggestion
** Features: The Azure subscription must be registered for the following feature.
```

File: modules/get-started/pages/cluster-types/byoc/create-byoc-cluster-azure.adoc [hunk 2]
```diff
@@ -16,7 +16,10 @@
...
++
+*** Microsoft.Compute/EncryptionAtHost. For more information, see the https://learn.microsoft.com/en-us/azure/virtual-machines/linux/disks-enable-host-based-encryption-cli#prerequisites[Microsoft documentation^].
```
Comment (asimms41) [inline modules/get-started/pages/cluster-types/byoc/create-byoc-cluster-azure.adoc:22]:
```suggestion
*** https://learn.microsoft.com/en-us/azure/virtual-machines/linux/disks-enable-host-based-encryption-cli#prerequisites[Microsoft.Compute/EncryptionAtHost].
```

Comment (asimms41) [inline modules/get-started/pages/cluster-types/byoc/create-byoc-cluster-azure.adoc:22]:
```suggestion
*** Microsoft.Compute/EncryptionAtHost.
```

**Comment by ligfx** [comment]
Thanks for the suggestions @asimms41 , I updated with the changes.

File: modules/get-started/pages/cluster-types/byoc/create-byoc-cluster-azure.adoc (hunk 1 above)
Comment (asimms41) [reply modules/get-started/pages/cluster-types/byoc/create-byoc-cluster-azure.adoc]: Updated

File: modules/get-started/pages/cluster-types/byoc/create-byoc-cluster-azure.adoc (hunk 2 above)
Comment (asimms41) [reply modules/get-started/pages/cluster-types/byoc/create-byoc-cluster-azure.adoc]: Updated

**Review by ncole — APPROVED** [review]

**Review by asimms41 — APPROVED** [review]
LGTM

**Pull request merged**